libmamba/src/core/environment.cpp

```diff
@@ -18,52 +21,86 @@ namespace mamba
         std::optional<std::string> get(const std::string& key)
         {
 #ifdef _WIN32
-            const size_t initial_size = 1024;
-            std::unique_ptr<char[]> temp_small = std::make_unique<char[]>(initial_size);
-            std::size_t size = GetEnvironmentVariableA(key.c_str(), temp_small.get(), initial_size);
-            if (size == 0)  // Error or empty/missing
+            // See:
+            // https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/getenv-s-wgetenv-s?view=msvc-170
+            static std::mutex call_mutex;
+            std::scoped_lock ready_to_execute{ call_mutex };  // Calls to getenv_s kinds of
+                                                              // functions are not thread-safe, this
+                                                              // is to prevent related issues.
+
+            const auto on_failed = [&](errno_t error_code)
+            {
+                LOG_ERROR << fmt::format(
+                    "Failed to acquire environment variable '{}' : errcode = {}",
+                    key,
+                    error_code
+                );
```

Comment:
Nit: One can also do (with the right header)
```suggestion
                fmt::format_to(LOG_ERROR,
                    "Failed to acquire environment variable '{}' : errcode = {}",
                    key,
                    error_code
                );
```

Reply:
BTW I know this is possible but I dont know if that changes something performance-wise for fmt? Do you know?

Reply:
Actually I tried that syntax and it doenst seem to compile, probably lacks some compatibility layer to add. I'll keep it as is for now.